visualization/app/codeCharta/state/metric.service.ts

```diff
@@ -63,14 +53,8 @@ export class MetricService implements FilesSelectionSubscriber, BlacklistSubscri
 	}
 
 	public getAttributeTypeByMetric(metricName: string, state: State): AttributeTypeValue {
-		const attributeType = this.getMergedAttributeTypes(state.fileSettings.attributeTypes).find(x => {
-			return _.findKey(x) === metricName
-		})
-
-		if (attributeType) {
-			return attributeType[metricName]
-		}
-		return null
+		const attributeType = state.fileSettings.attributeTypes.nodes[metricName]
+		return !!attributeType ? attributeType : null
```

Comment:
why do we return null here instead of undefined?

```
return attributeType
```

Reply:
So when I was refactoring the method, i included this line because there are tests that fail if the return value is undefined instead of null.

However, i checked it just now, and there seems to be no good reason why undefined should be a less desirable return value than null (so the test case was specified too strict probably). I will therefore simplify the function and adjust the tests.

Reply:
the only way to return null, if we actually read a cc.json that has `key: "rloc"`, `value: null`
and check for `state.fileSettings.attributeTypes.nodes["rloc"]

If I'm not mistaken, we can use this test instead:
```
it("should return undefined if attributeType not available", () => {
			const actual = metricService.getAttributeTypeByMetric("notfound", state)

			expect(actual).toBeUndefined()
		})
```

Reply:
Hmm this may get philosophical... so for our app we need that "actual" evaluates to false in an if statement. Therefore for any purposes in our app null or undefined are fine, which is why i have decided to test for !actual==true. (because the previous test, that required the result to be null made me think during refactoring that null is required when in fact it wasn't).

So from my point of view the question is: is there a value from defining the test as narrow as possible, even if the app does not require it?